Product filtering (using checkboxes for categories & reset)

harrydigos opened the pull request.

panteliselef commented on components/products/ProductFilter.tsx:
```diff
@@ -0,0 +1,67 @@
+
+  const onSuccessfulSubmit = (data: CheckFields) => {
+    let appliedFilters: string[] = [];
+    Object.entries(data).forEach(([key, value]) => (value ? appliedFilters.push(key) : null));
```
use .filter() for this

harrydigos replied:
Tried it but it didn't work as I wanted

panteliselef commented on components/products/ProductFilter.tsx:
```diff
@@ -0,0 +1,67 @@
+
+const checkboxes = ["Espresso", "Beans", "Pods", "Frothers", "Grinders", "Accessories", "Cups"] as const;
+
+type TupleToObject<T extends readonly string[]> = {
```
this is a utility type, it'd be nice to have it a different file

panteliselef commented on components/products/ProductFilter.tsx:
```diff
@@ -0,0 +1,67 @@
+  }
+);
+
+const checkboxes = ["Espresso", "Beans", "Pods", "Frothers", "Grinders", "Accessories", "Cups"] as const;
```
@BabisDigos  ❓why are those capitalised ?

harrydigos replied:
In DB I have them like that, and I was bored changing them.

harrydigos committed:
Filter UI, Checkboxes added

Checkbox interaction w/ hook form

Fetching when selecting filters from checkboxes


Improvements

panteliselef reviewed (approved):
💯

harrydigos committed:
Range Slider w/ radix


Product fetching w/ price & categories


Added provider for filters

+ Splitted Price filter 
Minor improvement
